pkg/addons/default/coredns.go

```diff
@@ -145,68 +147,73 @@ func InstallCoreDNS(rawClient kubernetes.RawClientInterface, region string, wait
 		}
 		watcher.Stop()
 		if timeout {
-			return fmt.Errorf("timed out (after %s) waiting for %q pods to become ready", waitTimeout, CoreDNS)
+			return false, fmt.Errorf("timed out (after %s) waiting for %q pods to become ready", waitTimeout, CoreDNS)
 		}
 	}
 
+	if plan {
```

Comment:
I would move this to line 108 so it's out of the way as soon as possible

Reply:
The idea is that we actually run up to the point where we have done everything, except for updating, so that we can see if all read calls will succeed...